ckanext/unaids/helpers.py

```diff
@@ -166,3 +170,20 @@ def get_google_analytics_id():
 
 def is_an_estimates_dataset(dataset_type_name):
     return 'estimates' in dataset_type_name.lower()
+
+
+def url_encode(url):
+    # Find out if the url contains http or https then put it at index 0 of the list
```

Comment:
You can use https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote here instead of building the encoding from scratch.
e.g:
```
>>> quote('https://adr.local/download/f%%.csv', safe=':/')
'https://adr.local/download/f%25%25.csv'
```